mautic/mautic
# Replacing deprecated method orX() with or()

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

We noticed that PHPSTAN was randomly failing CI builds with this error:
```
 ------ -------------------------------------------------------------------------------- 
  Line   UserRepository.php                                                              
 ------ -------------------------------------------------------------------------------- 
  289    Method Doctrine\ORM\Query\Expr::orX() invoked with 2 parameters, 0-1 required.  
 ------ -------------------------------------------------------------------------------- 
```
Since the `orX` method is deprecated this replaces it with method `or` which is a replacement and can accept multiple params.

The method accepts both ORM and DBAL builders. One of them prefers `or` and the other does not yet have the replacement for `orX` so we must ensure we call the right method for both builders. It will be resolved in Doctrine 2.11:

https://github.com/doctrine/dbal/pull/3851

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to the user list page
3. Search a user with the `name:` prefix following first or last name
